### ISSUE TITLE
followed instructions and 

### ISSUE THREAD

**Issue by exnfox** [AkashKarnatak/Ajnabee#5]
left at this :

~/Ajnabee$ npm run dev
\
> Ajnabee@1.0.0 dev
> nodemon app.js -e js,html,css

[nodemon] 2.0.22
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,html,css
[nodemon] starting `node app.js`
file:///home/dillon/Ajnabee/db/sqlite.js:29
await db.init()
^^^^^

SyntaxError: Unexpected reserved word
    at Loader.moduleStrategy (internal/modules/esm/translators.js:133:18)
[nodemon] app crashed - waiting for file changes before starting...

**Comment by lukeidiot** [comment]
The error message you're encountering, SyntaxError: Unexpected reserved word, suggests there's an issue with the use of await in your Node.js application. The await keyword can only be used inside an async function. Here are a few steps to troubleshoot and resolve this issue:

Check Function Context: Ensure that the await db.init() line is inside an async function. If it's not, you'll need to either move it into an async function or wrap it with one.

Correct Syntax: If the function containing await db.init() is intended to be asynchronous, make sure you've declared it with the async keyword. For example:

```
javascript
Copy code
async function someFunction() {
    await db.init();
    // other code
}
```
Top-Level Await (Node.js 14+): If you're using Node.js version 14 or newer, top-level await is supported in ES modules. If your file is a CommonJS module (.js extension without type: "module" in package.json), you won't be able to use top-level await. Ensure your file is an ES module by either renaming it with a .mjs extension or specifying "type": "module" in your package.json.

Check Node.js Version: Verify that your Node.js version supports the await keyword outside of functions (if that's what you're attempting). This feature is available in Node.js 14.3.0 and later versions.

Restart Nodemon: Sometimes, simply restarting Nodemon can resolve the issue if it's related to caching or similar problems.

Check for Typos: Ensure that there are no typos or syntax errors near the await db.init() line.

Check db.init Function: Make sure that the db.init function itself is declared correctly and, if it's supposed to be asynchronous, that it's marked with the async keyword.

Check Error Stack Trace: The stack trace might provide more specific information about where and why the error is occurring. Look at the lines of code mentioned in the stack trace for more clues.

If after these checks the issue still persists, you might need to share more code context or specific parts of your setup for a more detailed diagnosis.

**Comment by exnfox** [comment]
i ran the code as downloaded and i got that error you are saying it is in sql.db file?

**Comment by lukeidiot** [comment]
no im saying you are prob using wrong node version

**Comment by exnfox** [comment]
well i updated node js and now the server is running however the video /webcam never loads, doesnt even ask for permission or aqnything

**Comment by lightlessdays** [comment]
The webcam won't load until you host the backend on a remote server, like Vercel. Don't worry - it will work when you publish it.

**Comment by elglaoui69** [comment]
> The webcam won't load until you host the backend on a remote server, like Vercel. Don't worry - it will work when you publish it.

Hello
thanks for your job , but why its not working on local ?

**Comment by lightlessdays** [comment]
> > The webcam won't load until you host the backend on a remote server, like Vercel. Don't worry - it will work when you publish it.
> 
> Hello
> thanks for your job , but why its not working on local ? 


`getUserMedia` which is needed for camera access is not allowed on http (at least in Chrome since ver 47), so try to serve your site over https. localhost doesn't serve over https.

**Comment by anantkrsingh** [comment]
> The webcam won't load until you host the backend on a remote server, like Vercel. Don't worry - it will work when you publish it.

Does this work over vercel because vercel supports only serverless hosting

**Comment by lightlessdays** [comment]
Please read my last reply. You must have a valid SSL certificate (https) to view webcam.

**Comment by anantkrsingh** [comment]
> Please read my last reply. You must have a valid SSL certificate (https) to view webcam.

yeah i have valid ssl please view at mujomegle.com

**Comment by lightlessdays** [comment]
I checked - both camera and microphone on your site is working. @anantkrsingh

**Comment by anantkrsingh** [comment]
> I checked - both camera and microphone on your site is working. @anantkrsingh

yeah its working but when i am trying to connect from other pc browser it is not working websocket connection getting failed

**Comment by lightlessdays** [comment]
I tried with two PCs. I had no problems. Probably a problem with your browser.

**Comment by anantkrsingh** [comment]
> I tried with two PCs. I had no problems. Probably a problem with your browser.

but same ajnabee.live is running in that browser

**Comment by stray131** [comment]
> > > The webcam won't load until you host the backend on a remote server, like Vercel. Don't worry - it will work when you publish it.
> > 
> > 
> > Hello
> > thanks for your job , but why its not working on local ?
> 
> `getUserMedia` which is needed for camera access is not allowed on http (at least in Chrome since ver 47), so try to serve your site over https. localhost doesn't serve over https.

I have been trying all summer to get this to work on my server using wss and https and reverse proxy. for some reason I cannot get it to work. Neither can chatgpt. 
running apache, not njinx.

Ive been trying to get it to work by changing the env.js to 
export const WEBSOCKET_URL = "wss://mysite.com:8080/"
and reverse proxy in apache. 
the web browser says 
WebSocket connection to 'wss://beginnerchat.com:8443/' failed: 
(anonymous) | @ | socket.js:31
  | createSocket | @ | socket.js:30
  | (anonymous) | @ | video.js:11

**Comment by RagnarTheGreat** [comment]
> > I tried with two PCs. I had no problems. Probably a problem with your browser.
> 
> but same ajnabee.live is running in that browser

i see you got it working? i just visited your site can you share how you got it to work? im having the issue where 2 users can connect and chat but cant see each other cameras

**Comment by RagnarTheGreat** [comment]
or better question is how do i even deploy it with vercel? which files are the backend file. im a complete noob lol
